### PR TITLE
in_winlog: add a new plugin for Windows Event Log

### PR DESCRIPTION
**Pull request by fujimotos**
This adds basic support for Windows Event Log. This plugin can:

 1. Read and parse Event Log records using Win32 API.
 2. Subscribe to multiple log channels at once.
 3. Preserve the read offset between restart (using SQLite3)

The biggest limitation right now is that it uses non-Unicode functions.
I'm planning to add Unicode support in the subsequent PRs.

Signed-off-by: Fujimoto Seiji <fujimoto@clear-code.com>

### CODE REVIEW

**Comment by fujimotos** [comment]
Documentation: fluent/fluent-bit-docs/pull/176
Related issue: #960

File: plugins/in_winlog/in_winlog.c [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+{
+    int ret;
+    const char *tmp;
+    struct winlog_channel *ch;
+    struct flb_in_winlog_config *ctx;
+
+    /* Initialize context */
+    ctx = flb_calloc(1, sizeof(struct flb_in_winlog_config));
+    if (!ctx) {
+        perror("malloc");
```
Comment (edsiper) [inline plugins/in_winlog/in_winlog.c]: please use flb_errno()

File: plugins/in_winlog/in_winlog.c [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+    tmp = flb_input_get_property("interval_sec", in);
+    if (tmp != NULL && atoi(tmp) > 0) {
+        ctx->interval_sec = atoi(tmp);
+    }
+
+    /* Read Buffer */
+    ctx->bufsize = DEFAULT_BUFFER_SIZE;
+    ctx->buf = flb_malloc(ctx->bufsize);
+    if (!ctx->buf) {
+        perror("malloc");
```
Comment (edsiper) [inline plugins/in_winlog/in_winlog.c]: flb_errno();

File: plugins/in_winlog/winlog.h [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+#ifndef FLB_WINLOG_H
+#define FLB_WINLOG_H
+
+struct winlog_channel {
+    HANDLE h;
+    char *name;
+    unsigned int record_number;
+    unsigned int time_written;
+    unsigned int seek;
+    struct winlog_channel *next;
```
Comment (edsiper) [inline plugins/in_winlog/winlog.h]: Please do not use old-fashion linked list, use struct mk_list instead

**Comment by fujimotos** [comment]
Thank you for the code review! I'll follow them up next week.

File: plugins/in_winlog/in_winlog.c (hunk 1 above)
Comment (fujimotos) [reply plugins/in_winlog/in_winlog.c]: I fixed it to use `flb_errno()`.

File: plugins/in_winlog/in_winlog.c (hunk 2 above)
Comment (fujimotos) [reply plugins/in_winlog/in_winlog.c]: I fixed it to use `flb_errno()` too.

File: plugins/in_winlog/winlog.h (hunk 3 above)
Comment (fujimotos) [reply plugins/in_winlog/winlog.h]: Now it uses `struct mk_list` to manage multiple channels.

**Comment by fujimotos** [comment]
@edsiper I've fixed the code according to your suggestion.

The testing failure on AppVeyor is due to the recent changes introduced
into master, which I addressed in the following commits: 

* https://github.com/edsiper/chunkio/pull/42
* https://github.com/edsiper/chunkio/pull/44
* https://github.com/fluent/fluent-bit/pull/1395

With all of them merged, I can confirm that AppVeyor runs green again

**Comment by edsiper** [comment]
thanks

**Pull request merged**